File: Day18_Delete_Middle_of_LinkedList/delete_middle_linkedlist.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteMiddle(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        dummy=ListNode()
        dummy.next=head
        s=dummy
        f=head
        count=0
        while f is not None:
            f=f.next
            print(s.val)
            count+=1
            if(count%2==0):
                s=s.next
        s.next=s.next.next
        return dummy.next
        

File: Day18_Delete_Middle_of_LinkedList/delete_middle_linkedlist.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        ListNode dummy=new ListNode(0,head);
        ListNode s=dummy;
        ListNode f=head;
        int count=0;
        while(f!=null){
            f=f.next;
            count+=1;
            if(count%2==0){
                s=s.next;
            }
        }
        s.next=s.next.next;
        return dummy.next;

    }
}
